specs/vep-2448-vdk-run-logs-simplified-and-readable/README.md

```diff
@@ -366,10 +366,290 @@ or slowed down due to progress bar updates. This is really relevant for
 ingestion.
 
 Also as we need progress indicators in multiple places likely we need a common
-encapsulation for that - in a form of a python module or plugin. The progress
+encapsulation (or abstraction) for that - in a form of a python module or plugin. The progress
 indicator should provide a notification callback mechanism or similar so it's
 integratable with Notebook
 
+For more detailed understanding of the user workflows [this research documented is recommended](https://github.com/vmware/versatile-data-kit/wiki/Research:-Progress-Indicators)
+
+#### Technical Requirements
+
+* The progress abstraction should support multi step operations. Different steps are started and ended multiple times one after another in a job.
+* The abstraction should support hierarchical operations. A Step can start other long running operations (e.g a DAG sub-jobs)
+* The abstraction should support operations running concurrently (that is multiple tasks are started and keep progressing concurrently - in the same thread).
+  For example, this requirement comes primarily because of DAGs. Multiple sub-jobs started in a DAG can run concurrently (no multi-threading needed here though)
+* The abstraction should support operations running in background threads. When you send data for ingestion, that data is processed in background while the job continue with other operations. The progress track can be updated from multiple threads as they ingest data
+* The abstraction should allow for the total to change. E.g when we are ingesting data the total number of objects being ingested is changed as user adds new and new ones.
+* Performance overhead should be minimal as this may be called millions of times (e.g for each ingested row in a payload)
+* (Optional) The abstraction can be extended to collect metrics  about the state of operations running by the framework (bandwidth, duration, failed requests, etc).
```

Comment:
This is way out of scope in my opinion.

Reply:
Which one ? 

The last one or all selected? 

This last one is marked as optional because indeed  I think it's out of scope.  
When thinking of a solution I want to think about all the actual problems and needs we've observed that we need to cover even if we decide to implement only part of that. 

For the rest, Tracking ingestion is a rather important use case. 

Currently the way it's tracked is through rather very verbose logs like this: 

```
2023-08-04 10:22:04,608 [VDK] xxxx [INFO ] vdk.internal.builtin_plugins.i ingester_router.py:72 send_object_for_[id:aaf22f1f-8202-4155-b8b5-3d14ea0a5244-1691169635-97af1]- Sending object for ingestion with method: http and target: https://scapi-stg.vmware.com/sc/api/collectors/xxxx/instances/xxxx/batch?collection_id=xxx%7Caaf22f1f-8202-4155-234234-1691169635
2023-08-04 10:22:04,609 [VDK] historical-extrapolated
```

But those can be a lot , like literally way too much.  This causes the job to be very slow as at least one client complained during the user interviews.


We want to stop logging for each payload and keep track of the payloads smarter. Because The moment we remove those logs, people would complain that the program is stucked - although what it is doing is ingesting payloads.